src/DotVVM.Framework/Compilation/DefaultControlBuilderFactory.cs

```diff
@@ -68,18 +68,30 @@ private IControlBuilder CreateControlBuilder(MarkupFile file)
                 var namespaceName = GetNamespaceFromFileName(file.FileName, file.LastWriteDateTimeUtc);
                 var assemblyName = namespaceName;
                 var className = GetClassFromFileName(file.FileName) + "ControlBuilder";
-                try
-                {
-                    return ViewCompilerFactory().CompileView(file.ContentsReaderFactory(), file.FileName, assemblyName, namespaceName, className);
-                }
-                catch (DotvvmCompilationException ex)
+                void editCompilationException(DotvvmCompilationException ex)
```

Comment:
Local function in the middle of a function? Wouldn't it be better to leave there at the end of the method?